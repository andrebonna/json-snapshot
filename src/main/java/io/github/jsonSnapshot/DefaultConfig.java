package io.github.jsonSnapshot;

/** Alias for JUnitConfig for Backward Compatibility */
public class DefaultConfig extends JUnitConfig {}
