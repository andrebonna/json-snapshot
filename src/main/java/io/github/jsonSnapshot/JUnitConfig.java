package io.github.jsonSnapshot;

import java.lang.reflect.Method;
import java.util.stream.Stream;

import lombok.Getter;

public class JUnitConfig implements SnapshotConfig {
  @Getter private String filePath = "src/test/java/";

  @Override
  public StackTraceElement findStacktraceElement() {
    StackTraceElement[] stackTraceElements = Thread.currentThread().getStackTrace();
    int elementsToSkip = 1; // Start after stackTrace
    while (SnapshotMatcher.class
        .getName()
        .equals(stackTraceElements[elementsToSkip].getClassName())) {
      elementsToSkip++;
    }

    return Stream.of(stackTraceElements)
        .skip(elementsToSkip)
        .filter(
            stackTraceElement ->
                hasTestAnnotation(
                    SnapshotMatcher.getMethod(
                        getClassForName(stackTraceElement.getClassName()),
                        stackTraceElement.getMethodName())))
        .findFirst()
        .orElseThrow(
            () ->
                new SnapshotMatchException(
                    "Could not locate a method with one of supported test annotations"));
  }

  private boolean hasTestAnnotation(Method method) {
    return
    // Junit 4
    method.isAnnotationPresent(org.junit.Test.class)
        || method.isAnnotationPresent(org.junit.BeforeClass.class)
        // JUnit 5
        || method.isAnnotationPresent(org.junit.jupiter.params.ParameterizedTest.class)
        || method.isAnnotationPresent(org.junit.jupiter.api.Test.class)
        || method.isAnnotationPresent(org.junit.jupiter.api.BeforeAll.class);
  }

  private Class<?> getClassForName(String className) {
    try {
      return Class.forName(className);
    } catch (ClassNotFoundException e) {
      throw new IllegalArgumentException(e);
    }
  }
}
