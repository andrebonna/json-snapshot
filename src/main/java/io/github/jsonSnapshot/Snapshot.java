package io.github.jsonSnapshot;

import lombok.Getter;
import org.assertj.core.util.diff.DiffUtils;
import org.assertj.core.util.diff.Patch;

import java.lang.reflect.Method;
import java.util.Arrays;
import java.util.Collection;
import java.util.Map;
import java.util.Set;
import java.util.function.Function;

public class Snapshot {

    private SnapshotFile snapshotFile;

    private Class clazz;

    private Method method;

    private Function<Object, String> jsonFunction;

    @Getter
    private Object[] current;

    Snapshot(SnapshotFile snapshotFile, Class clazz, Method method, Function<Object, String> jsonFunction, Object... current) {
        this.current = current;
        this.snapshotFile = snapshotFile;
        this.clazz = clazz;
        this.method = method;
        this.jsonFunction = jsonFunction;
    }

    public void toMatchSnapshot() {

        Set<String> rawSnapshots = snapshotFile.getRawSnapshots();

        String rawSnapshot = getRawSnapshot(rawSnapshots);

        String currentObject = takeSnapshot();

        // Match Snapshot
        if (rawSnapshot != null) {
            if (!rawSnapshot.trim().equals(currentObject.trim())) {
                throw generateDiffError(rawSnapshot, currentObject);
            }
        }
        // Create New Snapshot
        else {
            snapshotFile.push(currentObject);
        }
    }

    private SnapshotMatchException generateDiffError(String rawSnapshot, String currentObject) {
        //compute the patch: this is the diffutils part
        Patch<String> patch =
                DiffUtils.diff(
                        Arrays.asList(rawSnapshot.trim().split("\n")),
                        Arrays.asList(currentObject.trim().split("\n")));
        String error = "Error on: \n" + currentObject.trim() + "\n\n" + patch.getDeltas().stream().map(delta -> delta.toString() + "\n").reduce(String::concat).get();
        return new SnapshotMatchException(error);
    }

    private String getRawSnapshot(Collection<String> rawSnapshots) {
        for (String rawSnapshot : rawSnapshots) {

            if (rawSnapshot.contains(getSnapshotName())) {
                return rawSnapshot;
            }
        }
        return null;
    }

    private String takeSnapshot() {
        return getSnapshotName() + jsonFunction.apply(current);
    }

    public String getSnapshotName() {
        return clazz.getName() + "." + method.getName() + "=";
    }

    /**
     * Ignore nested fields in the {@link #current} objects.
     *
     * @param pathsToIgnore Dot delimited paths to ignore within the {@link #current} objects. For example {@code nested.field}.
     * @return self
     * @throws SnapshotMatchException if the {@link #current} objects are not nested {@link Map}s.
     */
    public Snapshot ignoring(String... pathsToIgnore) throws SnapshotMatchException {
        for (String path : pathsToIgnore) {
            for (Object object : current) {
                SnapshotUtils.deleteNestedFieldFromMap(object, path);
            }
        }

        return this;
    }
}
