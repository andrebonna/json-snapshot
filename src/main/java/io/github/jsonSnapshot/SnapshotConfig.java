package io.github.jsonSnapshot;

import lombok.Builder;
import lombok.Getter;

@Builder
@Getter
public class SnapshotConfig {
  @Builder.Default private final String filePath = "src/test/java/";
}
