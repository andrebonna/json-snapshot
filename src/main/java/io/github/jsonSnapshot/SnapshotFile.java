package io.github.jsonSnapshot;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileOutputStream;
import java.io.FileReader;
import java.io.IOException;
import java.util.Set;
import java.util.TreeSet;
import java.util.stream.Collectors;
import java.util.stream.Stream;

import lombok.Getter;

import org.apache.commons.lang3.StringUtils;

public class SnapshotFile {

  private static final String SPLIT_STRING = "\n\n\n";

  private String fileName;

  @Getter private Set<String> rawSnapshots;

  SnapshotFile(String filePath, String fileName) throws IOException {

    this.fileName = filePath + fileName;

    StringBuilder fileContent = new StringBuilder();

    try (BufferedReader br = new BufferedReader(new FileReader(this.fileName))) {

      String sCurrentLine;

      while ((sCurrentLine = br.readLine()) != null) {
        fileContent.append(sCurrentLine + "\n");
      }

      String fileText = fileContent.toString();
      if (StringUtils.isNotBlank(fileText)) {
        rawSnapshots =
            Stream.of(fileContent.toString().split(SPLIT_STRING))
                .map(String::trim)
                .collect(Collectors.toCollection(TreeSet::new));
      } else {
        rawSnapshots = new TreeSet<>();
      }
    } catch (IOException e) {
      createFile(this.fileName);
      rawSnapshots = new TreeSet<>();
    }
  }

  private File createFile(String fileName) throws IOException {
    File file = new File(fileName);
    file.getParentFile().mkdirs();
    file.createNewFile();
    return file;
  }

  public void push(String snapshot) {

    rawSnapshots.add(snapshot);

    File file = null;

    try {
      file = createFile(fileName);
    } catch (IOException e) {
      e.printStackTrace();
    }

    try (FileOutputStream fileStream = new FileOutputStream(file, false)) {
      byte[] myBytes = StringUtils.join(rawSnapshots, SPLIT_STRING).getBytes();
      fileStream.write(myBytes);
    } catch (IOException e) {
      e.printStackTrace();
    }
  }
}
