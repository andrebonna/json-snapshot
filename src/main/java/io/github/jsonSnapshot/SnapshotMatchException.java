package io.github.jsonSnapshot;

public class SnapshotMatchException extends RuntimeException {

  public SnapshotMatchException(String message) {
    super(message);
  }

  public SnapshotMatchException(String message, Throwable cause) {
    super(message, cause);
  }
}
