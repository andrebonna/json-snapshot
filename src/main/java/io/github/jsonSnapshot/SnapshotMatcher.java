package io.github.jsonSnapshot;

import static org.assertj.core.util.Arrays.isNullOrEmpty;

import java.io.IOException;
import java.lang.reflect.Method;
import java.util.ArrayList;
import java.util.List;
import java.util.Optional;
import java.util.Set;
import java.util.function.Function;
import java.util.stream.Collectors;
import java.util.stream.Stream;

import org.apache.commons.lang3.ArrayUtils;
import org.apache.commons.lang3.StringUtils;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.core.PrettyPrinter;
import com.fasterxml.jackson.core.util.DefaultIndenter;
import com.fasterxml.jackson.core.util.DefaultPrettyPrinter;
import com.fasterxml.jackson.core.util.DefaultPrettyPrinter.Indenter;
import com.fasterxml.jackson.core.util.Separators;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.SerializationFeature;

public class SnapshotMatcher {

  private static Logger log = LoggerFactory.getLogger(SnapshotMatcher.class);

  private static Class clazz = null;
  private static SnapshotFile snapshotFile = null;
  private static List<Snapshot> calledSnapshots = new ArrayList<>();
  private static Function<Object, String> jsonFunction;
  private static SnapshotConfig config;

  public static void start() {
    start(new DefaultConfig(), defaultJsonFunction());
  }

  public static void start(SnapshotConfig config) {
    start(config, defaultJsonFunction());
  }

  public static void start(Function<Object, String> jsonFunction) {
    start(new DefaultConfig(), jsonFunction);
  }

  public static void start(SnapshotConfig config, Function<Object, String> jsonFunction) {
    SnapshotMatcher.jsonFunction = jsonFunction;
    SnapshotMatcher.config = config;
    try {
      StackTraceElement stackElement = config.findStacktraceElement();
      clazz = Class.forName(stackElement.getClassName());
      snapshotFile =
          new SnapshotFile(
              config.getFilePath(), stackElement.getClassName().replaceAll("\\.", "/") + ".snap");
    } catch (ClassNotFoundException | IOException e) {
      throw new SnapshotMatchException(e.getMessage());
    }
  }

  public static void validateSnapshots() {
    Set<String> rawSnapshots = snapshotFile.getRawSnapshots();
    List<String> snapshotNames =
        calledSnapshots.stream().map(Snapshot::getSnapshotName).collect(Collectors.toList());
    List<String> unusedRawSnapshots = new ArrayList<>();

    for (String rawSnapshot : rawSnapshots) {
      boolean foundSnapshot = false;
      for (String snapshotName : snapshotNames) {
        if (rawSnapshot.contains(snapshotName)) {
          foundSnapshot = true;
        }
      }
      if (!foundSnapshot) {
        unusedRawSnapshots.add(rawSnapshot);
      }
    }
    if (unusedRawSnapshots.size() > 0) {
      log.warn(
          "All unused Snapshots: "
              + StringUtils.join(unusedRawSnapshots, "\n")
              + ". Consider deleting the snapshot file to recreate it!");
    }
  }

  public static Snapshot expectScenario(String scenario, Object firstObject, Object... others) {
    return expectCondition(scenario, firstObject, others);
  }

  public static Snapshot expect(Object firstObject, Object... others) {
    return expectCondition(null, firstObject, others);
  }

  /**
   * @param scenario for parameterized tests supply a unique scenario for each iteration
   * @param firstObject
   * @param others
   * @return
   */
  private static Snapshot expectCondition(String scenario, Object firstObject, Object... others) {

    if (clazz == null) {
      throw new SnapshotMatchException(
          "SnapshotTester not yet started! Start it on @BeforeClass/@BeforeAll with SnapshotMatcher.start()");
    }
    Object[] objects = mergeObjects(firstObject, others);
    StackTraceElement stackElement = config.findStacktraceElement();
    Method method = getMethod(clazz, stackElement.getMethodName());
    Snapshot snapshot =
        new Snapshot(config, snapshotFile, clazz, method, scenario, jsonFunction, objects);
    validateExpectCall(snapshot);
    calledSnapshots.add(snapshot);
    return snapshot;
  }

  static Function<Object, String> defaultJsonFunction() {

    ObjectMapper objectMapper = buildObjectMapper();

    PrettyPrinter pp = buildDefaultPrettyPrinter();

    return (object) -> {
      try {
        return objectMapper.writer(pp).writeValueAsString(object);
      } catch (Exception e) {
        throw new SnapshotMatchException(e.getMessage());
      }
    };
  }

  private static PrettyPrinter buildDefaultPrettyPrinter() {
    DefaultPrettyPrinter pp =
        new DefaultPrettyPrinter("") {
          @Override
          public DefaultPrettyPrinter withSeparators(Separators separators) {
            this._separators = separators;
            this._objectFieldValueSeparatorWithSpaces =
                separators.getObjectFieldValueSeparator() + " ";
            return this;
          }
        };
    Indenter lfOnlyIndenter = new DefaultIndenter("  ", "\n");
    pp.indentArraysWith(lfOnlyIndenter);
    pp.indentObjectsWith(lfOnlyIndenter);
    return pp;
  }

  private static ObjectMapper buildObjectMapper() {
    ObjectMapper objectMapper = new ObjectMapper();
    objectMapper.configure(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS, true);
    objectMapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);

    objectMapper.setVisibility(
        objectMapper
            .getSerializationConfig()
            .getDefaultVisibilityChecker()
            .withFieldVisibility(JsonAutoDetect.Visibility.ANY)
            .withGetterVisibility(JsonAutoDetect.Visibility.NONE)
            .withSetterVisibility(JsonAutoDetect.Visibility.NONE)
            .withCreatorVisibility(JsonAutoDetect.Visibility.NONE));
    return objectMapper;
  }

  private static void validateExpectCall(Snapshot snapshot) {
    for (Snapshot eachSnapshot : calledSnapshots) {
      if (eachSnapshot.getSnapshotName().equals(snapshot.getSnapshotName())) {
        throw new SnapshotMatchException(
            "You can only call 'expect' once per test method. Try using array of arguments on a single 'expect' call");
      }
    }
  }

  public static Method getMethod(Class<?> clazz, String methodName) {
    try {
      return Stream.of(clazz.getDeclaredMethods())
          .filter(method -> method.getName().equals(methodName))
          .findFirst()
          .orElseThrow(() -> new NoSuchMethodException("Not Found"));
    } catch (NoSuchMethodException e) {
      return Optional.ofNullable(clazz.getSuperclass())
          .map(superclass -> getMethod(superclass, methodName))
          .orElseThrow(
              () ->
                  new SnapshotMatchException(
                      "Could not find method "
                          + methodName
                          + " on class "
                          + clazz
                          + "\nPlease annotate your test method with @Test and make it without any parameters!"));
    }
  }

  private static Object[] mergeObjects(Object firstObject, Object[] others) {
    Object[] objects = new Object[1];
    objects[0] = firstObject;
    if (!isNullOrEmpty(others)) {
      objects = ArrayUtils.addAll(objects, others);
    }
    return objects;
  }
}
