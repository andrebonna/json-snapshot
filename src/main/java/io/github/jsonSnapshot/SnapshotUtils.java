package io.github.jsonSnapshot;

import static org.mockito.Mockito.atLeastOnce;
import static org.mockito.Mockito.verify;

import java.lang.reflect.Parameter;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.function.Function;

import org.apache.commons.lang3.ArrayUtils;
import org.assertj.core.util.Arrays;
import org.mockito.ArgumentCaptor;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.core.PrettyPrinter;
import com.fasterxml.jackson.core.util.DefaultIndenter;
import com.fasterxml.jackson.core.util.DefaultPrettyPrinter;
import com.fasterxml.jackson.core.util.Separators;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.SerializationFeature;

public class SnapshotUtils {

  public static <T> HashMap<String, List<LinkedHashMap<String, Object>>> extractArgs(
      T object, String methodName, SnapshotCaptor... snapshotCaptors) {
    List<ArgumentCaptor> captors = new ArrayList<>();
    Class[] classes = new Class[snapshotCaptors.length];

    int i = 0;
    for (SnapshotCaptor snapshotCaptor : snapshotCaptors) {
      classes[i] = snapshotCaptor.getParameterClass();
      captors.add(ArgumentCaptor.forClass(snapshotCaptor.getParameterClass()));
      i++;
    }

    return process(object, methodName, captors, classes, snapshotCaptors);
  }

  public static <T> HashMap<String, List<LinkedHashMap<String, Object>>> extractArgs(
      T object, String methodName, Class<?>... classes) {
    List<ArgumentCaptor> captors = new ArrayList<>();

    for (Class clazz : classes) {
      captors.add(ArgumentCaptor.forClass(clazz));
    }

    return process(object, methodName, captors, classes, null);
  }

  private static <T> HashMap<String, List<LinkedHashMap<String, Object>>> process(
      T object,
      String methodName,
      List<ArgumentCaptor> captors,
      Class[] classes,
      SnapshotCaptor[] snapshotCaptors) {
    HashMap<String, List<LinkedHashMap<String, Object>>> result = new HashMap<>();
    try {
      Parameter[] parameters =
          object.getClass().getDeclaredMethod(methodName, classes).getParameters();

      object
          .getClass()
          .getDeclaredMethod(methodName, classes)
          .invoke(
              verify(object, atLeastOnce()),
              captors.stream().map(ArgumentCaptor::capture).toArray());

      List<LinkedHashMap<String, Object>> extractedObjects = new ArrayList<>();

      int numberOfCall;

      if (captors.size() > 0) {
        numberOfCall = captors.get(0).getAllValues().size();

        for (int i = 0; i < numberOfCall; i++) {
          LinkedHashMap<String, Object> objectMap = new LinkedHashMap<>();

          int j = 0;
          for (ArgumentCaptor captor : captors) {
            Object value = captor.getAllValues().get(i);
            if (snapshotCaptors != null) {
              value = snapshotCaptors[j].removeIgnored(value);
            }
            objectMap.put(parameters[j].getName(), value);
            j++;
          }
          extractedObjects.add(objectMap);
        }
      }

      result.put(
          object.getClass().getSuperclass().getSimpleName() + "." + methodName, extractedObjects);
    } catch (Exception e) {
      throw new SnapshotMatchException(e.getMessage(), e.getCause());
    }

    return result;
  }

  public static Function<Object, String> defaultJsonFunction() {

    ObjectMapper objectMapper = buildObjectMapper();

    PrettyPrinter pp = buildDefaultPrettyPrinter();

    return (object) -> {
      try {
        return objectMapper.writer(pp).writeValueAsString(object);
      } catch (Exception e) {
        throw new SnapshotMatchException(e.getMessage());
      }
    };
  }

  public static PrettyPrinter buildDefaultPrettyPrinter() {
    DefaultPrettyPrinter pp =
        new DefaultPrettyPrinter("") {
          @Override
          public DefaultPrettyPrinter withSeparators(Separators separators) {
            this._separators = separators;
            this._objectFieldValueSeparatorWithSpaces =
                separators.getObjectFieldValueSeparator() + " ";
            return this;
          }
        };
    DefaultPrettyPrinter.Indenter lfOnlyIndenter = new DefaultIndenter("  ", "\n");
    pp.indentArraysWith(lfOnlyIndenter);
    pp.indentObjectsWith(lfOnlyIndenter);
    return pp;
  }

  private static ObjectMapper buildObjectMapper() {
    ObjectMapper objectMapper = new ObjectMapper();
    objectMapper.configure(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS, true);
    objectMapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);

    objectMapper.setVisibility(
        objectMapper
            .getSerializationConfig()
            .getDefaultVisibilityChecker()
            .withFieldVisibility(JsonAutoDetect.Visibility.ANY)
            .withGetterVisibility(JsonAutoDetect.Visibility.NONE)
            .withSetterVisibility(JsonAutoDetect.Visibility.NONE)
            .withCreatorVisibility(JsonAutoDetect.Visibility.NONE));
    return objectMapper;
  }

  public static void validateExpectCall(Snapshot snapshot, List<Snapshot> calledSnapshots) {
    for (Snapshot eachSnapshot : calledSnapshots) {
      if (eachSnapshot.getSnapshotName().equals(snapshot.getSnapshotName())) {
        throw new SnapshotMatchException(
            "You can only call 'expect' once per test method. Try using array of arguments on a single 'expect' call");
      }
    }
  }

  public static Object[] mergeObjects(Object firstObject, Object[] others) {
    Object[] objects = new Object[1];
    objects[0] = firstObject;
    if (!Arrays.isNullOrEmpty(others)) {
      objects = ArrayUtils.addAll(objects, others);
    }
    return objects;
  }
}
