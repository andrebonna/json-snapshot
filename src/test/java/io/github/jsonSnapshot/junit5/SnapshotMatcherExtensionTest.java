package io.github.jsonSnapshot.junit5;

import static io.github.jsonSnapshot.junit5.SnapshotMatcher.expect;
import static org.junit.jupiter.api.Assertions.assertThrows;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.ValueSource;

import io.github.jsonSnapshot.SnapshotMatchException;

public class SnapshotMatcherExtensionTest {

  @DisplayName("Test update events are processed")
  @ParameterizedTest(name = "test{0}")
  @ValueSource(strings = {"one", "two", "three", "four", "five", "six", "seven"})
  @ExtendWith(SnapshotMatcherExtension.class)
  public void testParameterizedTest(String param) {
    expect(param, param.toUpperCase()).toMatchSnapshot();
  }

  @DisplayName("Test update events are processed2")
  @ParameterizedTest(name = "test{0}")
  @ValueSource(strings = {"one", "two", "three", "four", "five", "six", "seven"})
  @ExtendWith(SnapshotMatcherExtension.class)
  public void testParameterizedTest2(String param) {
    expect(param, param.toUpperCase()).toMatchSnapshot();
  }

  @Test
  public void testCallsToSnapshotMatcherFailWhenExtensionIsNotPresent() {
    assertThrows(SnapshotMatchException.class, () -> expect("uh oh").toMatchSnapshot());
    assertThrows(SnapshotMatchException.class, () -> SnapshotMatcher.start());
    assertThrows(SnapshotMatchException.class, () -> SnapshotMatcher.validateSnapshots());
  }
}
