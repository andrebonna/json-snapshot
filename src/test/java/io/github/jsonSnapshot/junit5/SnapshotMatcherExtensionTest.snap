io.github.jsonSnapshot.junit5.SnapshotMatcherExtensionTest.testParameterizedTest.testfive=[
  "five",
  "FIVE"
]


io.github.jsonSnapshot.junit5.SnapshotMatcherExtensionTest.testParameterizedTest.testfour=[
  "four",
  "FOUR"
]


io.github.jsonSnapshot.junit5.SnapshotMatcherExtensionTest.testParameterizedTest.testone=[
  "one",
  "ONE"
]


io.github.jsonSnapshot.junit5.SnapshotMatcherExtensionTest.testParameterizedTest.testseven=[
  "seven",
  "SEVEN"
]


io.github.jsonSnapshot.junit5.SnapshotMatcherExtensionTest.testParameterizedTest.testsix=[
  "six",
  "SIX"
]


io.github.jsonSnapshot.junit5.SnapshotMatcherExtensionTest.testParameterizedTest.testthree=[
  "three",
  "THREE"
]


io.github.jsonSnapshot.junit5.SnapshotMatcherExtensionTest.testParameterizedTest.testtwo=[
  "two",
  "TWO"
]


io.github.jsonSnapshot.junit5.SnapshotMatcherExtensionTest.testParameterizedTest2.testfive=[
  "five",
  "FIVE"
]


io.github.jsonSnapshot.junit5.SnapshotMatcherExtensionTest.testParameterizedTest2.testfour=[
  "four",
  "FOUR"
]


io.github.jsonSnapshot.junit5.SnapshotMatcherExtensionTest.testParameterizedTest2.testone=[
  "one",
  "ONE"
]


io.github.jsonSnapshot.junit5.SnapshotMatcherExtensionTest.testParameterizedTest2.testseven=[
  "seven",
  "SEVEN"
]


io.github.jsonSnapshot.junit5.SnapshotMatcherExtensionTest.testParameterizedTest2.testsix=[
  "six",
  "SIX"
]


io.github.jsonSnapshot.junit5.SnapshotMatcherExtensionTest.testParameterizedTest2.testthree=[
  "three",
  "THREE"
]


io.github.jsonSnapshot.junit5.SnapshotMatcherExtensionTest.testParameterizedTest2.testtwo=[
  "two",
  "TWO"
]